### ISSUE TITLE
treetagger.fileExtension only works on export

### ISSUE THREAD

**Issue by amir-zeldes** [korpling/pepperModules-TreetaggerModules#2]
The property treetagger.fileExtension is used to determine the extension for a generated TreeTagger format file from SNP. It would be great to apply it to the importer as well, allowing the import of other extensions.

**Comment by amir-zeldes** [comment]
@FlorianZipser @thomaskrause I'm running into trouble adding this property for the importer. It doesn't seem to be possible to call getProperties inside the TreetaggerImporter's constructor: even after setProperties, the properties are not read. I can access the properties fine inside createPepperMapper, but that only gets triggered if there is a document to read, which has to have one of the hard-wired extensions (tt, tab, etc.). If all documents have a custom extension, I can't get createPepperMapper to trigger, which is the earliest point I get access to the module properties.

What's the canonical way to access module properties inside the importer's constructor, or at least before the mapper is created? These options don't work in the importer constructor:

`String extension = ((TreetaggerImporterProperties) super.getProperties()).getImportFileExtension();`

Or:

`String extension = getProperties().getProperties().getProperty("treetagger.fileExtension");`

**Comment by FlorianZipser** [comment]
Yes, that's true, You cannot access the properties in constructor. This applies not only to properties, but for all fields, since they cannot be set before the object was created. So Pepper needs to create an object first before it can set its members. 
There are some places access the properties before a mapper is created. In your case I guess the best place would be to have something like this:

``` java
@Override
public void importCorpusStructure(SCorpusGraph corpusGraph){
   getDocumentEndings().add(getProperties()).getImportFileExtension());
   super.importCorpusStructure(corpusGraph);
}
```
